homeassistant/components/huisbaasje/sensor.py

```diff
@@ -0,0 +1,255 @@
+"""Platform for sensor integration."""
+from datetime import timedelta
+import logging
+
+import async_timeout
+from huisbaasje import HuisbaasjeException
+from huisbaasje.const import (
+    SOURCE_TYPE_ELECTRICITY,
+    SOURCE_TYPE_ELECTRICITY_IN,
+    SOURCE_TYPE_ELECTRICITY_IN_LOW,
+    SOURCE_TYPE_ELECTRICITY_OUT,
+    SOURCE_TYPE_ELECTRICITY_OUT_LOW,
+    SOURCE_TYPE_GAS,
+)
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_ID,
+    DEVICE_CLASS_POWER,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_WATT,
+    VOLUME_CUBIC_METERS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    DOMAIN,
+    FLOW_CUBIC_METERS_PER_HOUR,
+    POLLING_INTERVAL,
+    SENSOR_TYPE_RATE,
+    SENSOR_TYPE_THIS_DAY,
+    SOURCE_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, config_entry: ConfigEntry, async_add_entities
+):
+    """Set up the sensor platform."""
+    user_id = config_entry.data[CONF_ID]
+    # Get the Huisbaasje client
+    huisbaasje = hass.data[DOMAIN][user_id]
+
+    async def async_update_data():
+        return await async_update_huisbaasje(huisbaasje)
+
+    # Create a coordinator for polling updates
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name="sensor",
+        update_method=async_update_data,
+        update_interval=timedelta(seconds=POLLING_INTERVAL),
+    )
+
+    # Fetch initial data
+    await coordinator.async_refresh()
+
+    async_add_entities(
+        [
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity Rate",
+                device_class=DEVICE_CLASS_POWER,
+                source_type=SOURCE_TYPE_ELECTRICITY,
+            ),
+            HuisbaasjeSensor(
+                coordinator,
+                user_id=user_id,
+                name="Electricity In Rate",
```

Comment:
But is is not cumulative, it is the current rate (in Watt) if electricity. Maybe you mean "Electricity Today" to be called "Cumulative Energy"?

Reply:
Ah OK, rate means flow: electricity-flow in W or gas-flow in m3/hr.
Then I'm with @wouterbaake : "Power .." or "Current Power .." instead of "Electricity .. Rate".
And e.g. "Electricity Today" or "Energy Today" for the kWh-measurement so that it's clear that it's about energy-consumption during a time-period. And if you don't like to mention the time-period I'd suggest using the word "Cumulative".

Reply:
I have updated it to "Current power" and "Energy today" (and moved it to const.py)